relay-general/src/pii/processor.rs

```diff
@@ -373,9 +388,58 @@ mod tests {
         LogEntry, NativeDebugImage, Request, Span, TagEntry, Tags,
     };
     use crate::testutils::assert_annotated_snapshot;
-    use crate::types::{Annotated, Object, Value};
+    use crate::types::{Annotated, FromValue, Object, Value};
+
+    fn to_pii_config(datascrubbing_config: &DataScrubbingConfig) -> Option<PiiConfig> {
+        use crate::pii::convert::to_pii_config as to_pii_config_impl;
+        let rv = to_pii_config_impl(datascrubbing_config).unwrap();
+        if let Some(ref config) = rv {
+            let roundtrip: PiiConfig =
+                serde_json::from_value(serde_json::to_value(config).unwrap()).unwrap();
+            assert_eq!(&roundtrip, config);
+        }
+        rv
+    }
 
-    use super::*;
+    #[test]
+    fn test_scrub_original_value() {
+        let mut data = Event::from_value(
+            serde_json::json!({
+                "user": {
+                    "username": "hey  man 73.133.27.120", // should be stripped despite not being "known ip field"
+                    "ip_address": "is this an ip address? 73.133.27.120", //  <--------
+                },
+                "breadcrumbs": {
+                    "values": [
+                        {
+                            "message": "73.133.27.120",
+                            "data": {
+                                "test_data": "73.133.27.120" // test deep wildcard stripping
+                                }
+                        },
+                    ],
+                },
+                "sdk": {
+                    "client_ip": "should also be stripped"
+                },
+            })
+            .into(),
+        );
+
+        let scrubbing_config = DataScrubbingConfig {
+            scrub_data: false,
+            scrub_ip_addresses: true,
+            scrub_defaults: false,
+            ..Default::default()
+        };
+
+        let pii_config = to_pii_config(&scrubbing_config).unwrap();
+        let mut pii_processor = PiiProcessor::new(pii_config.compiled());
+
+        process_value(&mut data, &mut pii_processor, ProcessingState::root()).unwrap();
+
+        assert_debug_snapshot!(&data);
+    }
```

Comment:
Please also add a test where the the original value is a credit card number, so we can see if it is properly scrubbed.